### DISCUSSION TITLE
Losing internet connectivity causes stutter

### DISCUSSION THREAD

**Discussion by Syncrossus** [ppy/osu#17695] [Q&A]
### Type

Performance

### Bug description

Losing internet connectivity during gameplay causes momentary stuttering (at least on my system, which runs Linux).
Steps to reproduce:

1. Start a game
2. Cut the internet connection

This may or may not be related to #16466. Having looked at their network log, while they do not have the `TimeoutException`s and that I find in my logs, they do have many other network-related exceptions.
This also may or may not be related to issue #16822 which has been resolved but reported crashes when connectivity was lost in multiplayer.
If fixing the performance issue is difficult, one possibility may be to have an online/offline toggle in the options.

### Screenshots or videos

Video in which I purposefully cut my internet connection and the game instantly stutters:

https://user-images.githubusercontent.com/12431317/162081835-d70ecd22-6671-4992-9e99-7e06816d6fe8.mp4

### Version

2022.327.0 ; 2022.405.0

### Logs

[network.log](https://github.com/ppy/osu/files/8433272/network.log)
[runtime.log](https://github.com/ppy/osu/files/8433273/runtime.log)
[performance.log](https://github.com/ppy/osu/files/8433274/performance.log)
[database.log](https://github.com/ppy/osu/files/8433275/database.log)

**Comment by smoogipoo** [comment]
This is going to be something with your system and up to you to resolve, as there's no stutter in the frame time graphs and I also can't repro this.
